src/test/unit/base/services/authenticationService.test.ts

```diff
@@ -71,48 +71,22 @@ suite('AuthenticationService', () => {
   test('getIpFamily returns undefined when IPv6 not supported', async () => {
     const ipv6ErrorCode = 'EADDRNOTAVAIL';
 
-    // code-client calls 'needle', thus it's the easiest place to stub the response when IPv6 is not supported by the OS network stack. Otherwise, Node internals must be stubbed to return the error.
-    sinon.stub(needle, 'request').callsFake((_, uri, data, opts, callback) => {
-      if (!callback) throw new Error();
+    sinon.stub(dns, 'lookup').callsFake((_hostname, callback) => {
       callback(
         {
           code: ipv6ErrorCode,
           errno: ipv6ErrorCode,
         } as unknown as Error,
-        {} as unknown as NeedleResponse,
-        null,
+        '',
+        6,
       );
-      // eslint-disable-next-line camelcase
-      return needle.post(uri, data, { ...opts, ...overrideNeedleTimeoutOptions });
     });
 
     const ipFamily = await getIpFamily('https://dev.snyk.io');
 
     strictEqual(ipFamily, undefined);
   });
 
-  test('getIpFamily returns 6 when IPv6 supported', async () => {
```

Comment:
I think you could have kept this test and changed it to be for your new `getIpFamily` function to show backwards compatibility

Reply:
I'm wondering if we still need the other test though. wdyt?

Reply:
This one https://github.com/snyk/vscode-extension/blob/f3b8e34d5e26e4ae8c3eabe74b11399ba0ce8274/src/test/unit/base/services/authenticationService.test.ts#L74? I agree with you that before we shouldn't have really tested this function here, it should have been in `snyk/code-client`. But now that you implemented that function in the extension then it makes sense to keep both tests